pkg/testing/tools/artifacts_api.go

```diff
@@ -161,6 +170,62 @@ func (aac ArtifactAPIClient) GetVersions(ctx context.Context) (list *VersionList
 	return checkResponseAndUnmarshal[VersionList](resp)
 }
 
+// RemoveUnreleasedVersions from the list
+// There is a period of time when a version is already marked as released
+// but not published on the CDN. This happens when we already have build candidates.
+// This function checks if a version marked as released actually has published artifacts.
+// If there are no published artifacts, the version is removed from the list.
+func (aac ArtifactAPIClient) RemoveUnreleasedVersions(ctx context.Context, vList *VersionList) error {
+	suffix, err := testing.GetPackageSuffix(runtime.GOOS, runtime.GOARCH)
+	if err != nil {
+		return fmt.Errorf("failed to generate the artifact suffix: %w", err)
+	}
+
+	results := make([]string, 0, len(vList.Versions))
+
+	for _, versionItem := range vList.Versions {
+		parsedVersion, err := version.ParseVersion(versionItem)
+		if err != nil {
+			return fmt.Errorf("failed to parse version %s: %w", versionItem, err)
+		}
+		// we check only release versions without `-SNAPSHOT`
+		if parsedVersion.Prerelease() != "" {
+			results = append(results, versionItem)
+			continue
+		}
+		url := fmt.Sprintf("%s/elastic-agent-%s-%s", aac.cdnURL, versionItem, suffix)
```

Comment:
Will this work the same with the new agent release that will be available out of band thanks to @pchila's work?

Reply:
I can't confirm that, not familiar with the spec/implementation details but I'll talk to @pchila and I've assigned him as a reviewer.

Reply:
I don't know details yet on how the new `elastic-agent-x.y.z+buildYYYYMMDDHHMMSS ` are gonna be distributed. If Artifact API has those versions correctly returned in the version list and the *same exact version* appear in the CDN URL this could work otherwise this check will always fail for the early releases